syntax = "proto3";

package api;

import "pb/shared/content.proto";

service Chat {
    
  rpc Chat(ChatRequest) returns (stream ChatResponse) {}

}

message ChatRequest {
  repeated shared.Message messages = 1;
  repeated string models = 2;
  optional uint32 max_tokens = 3;
  repeated ToolDefinition tools = 4;
}

message ToolDefinition {
  string name = 1;
  string description = 2;
  string parameters_json = 3;  // JSON schema for parameters
}

message ChatResponse {
  // request id
  string id = 1;
  uint64 timestamp = 2;
  string model = 3;
  shared.Delta delta = 4;
}
